src/libos/src/signal/do_sigreturn.rs

```diff
@@ -216,14 +217,22 @@ fn handle_signals_by_user(
                     }
                 }
             }
-            const BIG_ENOUGH_GAP: u64 = 1024;
-            let stack_top = (curr_user_ctxt.gp_regs.rsp - BIG_ENOUGH_GAP) as usize;
+            const STACK_ALIGNMENT: usize = 16;
+            const RED_ZONE: u64 = 128;
+            let stack_top = align_down(
+                (curr_user_ctxt.gp_regs.rsp - RED_ZONE).try_into().unwrap(),
+                STACK_ALIGNMENT,
```

Comment:
Using `RED_ZONE` is a nice touch, but the alignment requirement is not necessary and kind of misleading, as it gives a false sense of safety.

>        // The x86 calling convention requires rsp to be 16-byte aligned.
>        // The following allocation on stack is right before we "call" the
>        // signal handler. So we need to make sure the allocation is at least
>        // 16-byte aligned.

As stated in the comments later, the 16-byte alignment is required by the calling convention, not arbitrary x86 instructions. We only need to ensure rsp is 16-byte aligned at the entry of a function. Having a 16-byte aligned `stack_stop` at this point is useless and gives a false sense of safety to readers, since we are going to push some more data structures on the stack before making a function call.

Even if the 16-byte alignment is desired, why not enforce the property in `Stack::new`? Enforcing the alignment there is much more reliable and we can make it an explicit property of a stack.